how-to/business-logic/how-to-change-user-role.md

```diff
@@ -0,0 +1,82 @@
+---
+title: How to Change User Role Programatically
+description: An example on changing the user role programatically, using cloud code
+type: how-to
+page_title: How to Change User Role Programatically
+slug: bl-change-user-role
+position:
+tags: backendservices, business-logic, cloud-code, user-role, cloud-functions
+teampulseid: 
+ticketid: 1016776
+pitsid: 
+---
+
+## Environment
+<table>
+  <tr>
+    <td>Service</td>
+    <td>Business Logic</td>
+  </tr>
+  <tr>
+    <td>Feature</td>
+    <td>Cloud Functions</td>
+  </tr>
+</table>
+
+## Description
+
+You need to change/set a user's role programatically based on a certain logic (without going to {{site.tp}} portal).
+
+## Solution
+
+You can set up a Cloud Function to update the user role. Updating a user role requires the use of a [Master Key]({http://docs.telerik.com/platform/backend-services/javascript/security/security-master-key-auth}) - it is strongly recommended not to include your Master Key inside your app code - therefore a perfect solution is to use clode code as the code stays on {{site.tp}} server. 
+
+A sample function in cloud code to change the user role would be:
+
+```JavaScript
+// In Cloud Function or Cloud Code for Data
+=========
+    // Set the new user role (the roleID of your custom role)
+    // Get the userId to which the new role will be applied.
+    var customRoleId = 'custom-role-id-here';
+    var userId = request.queryString.userId;
+
+    // Check if the user is eligible for role change (matches a condition you have set) 
+    if ('your-condition') {

```

Comment:
Eventually we may need to remove this check if it is explained in the Solution section